syntax = "proto3";

package mapreduce;

// Initial functions for the client --> master call
service MapReduce {
    rpc SubmitJob(JobRequest) returns (JobResponse);
}

// job_name is for the type of job if we add more in the future
// input_file is pretty straight forward
message JobRequest {
    string job_name = 1;
    string input_file = 2;
}

// message is if whether the job is accepted or an error occured
// success is the result of the job
// job_id is to separate each job from one another 
message JobResponse {
    string message = 1;
    bool success = 2;
    string job_id = 3;
}